metrics/storage/blogbench.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Description of the test:
+# This test runs the 'blogbench', and extracts the 'scores' for reads
+# and writes
+# Note - the scores are *not* normalised for the number of iterations run,
+# they are total scores for all iterations (this is the blogbench default output)
+
+set -e
+
+# General env
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+TEST_ARGS="$@"
+
+TEST_NAME="blogbench"
+IMAGE="local-blogbench"
+DOCKERFILE="${SCRIPT_PATH}/blogbench_dockerfile/Dockerfile"
+
+# Number of iterations for blogbench to run - note, results are not
+# scaled to iterations - more iterations results in bigger results
+ITERATIONS="${ITERATIONS:-30}"
+
+# Directory to run the test on
+TESTDIR="${TESTDIR:-/tmp}"
+CMD="blogbench -i ${ITERATIONS} -d ${TESTDIR}"
+
+function main() {
+	cmds=("awk")
+
+	init_env
+	check_cmds "${cmds[@]}"
+	check_dockerfiles_images "$IMAGE" "$DOCKERFILE"
+
+	metrics_json_init
+
+	local output=$(docker run --rm --runtime=$RUNTIME $IMAGE $CMD)
+
+	metrics_json_start_array
+
+	local writes=$(tail -2 <<< "$output" | head -1 | awk '{print $5}')
+	local reads=$(tail -1 <<< "$output" | awk '{print $6}')
+
+	local json="$(cat << EOF
+	{
+		"write": {
+			"Result" : $writes,
+			"Units"  : "items"
+		},
+		"read": {
+			"Result" : $reads,
+			"Units"  : "items"
+		}
+	}
+EOF
+)"
+
+	metrics_json_add_array_element "$json"
+	metrics_json_end_array "Results"
+	metrics_json_save
+	clean_env
```

Comment:
Does this function remove the created tmpfiles?

Reply:
@jcvenegas the tmpfile is created in the container